functions/go/create-setters/createsetters/create_setters.go

```diff
@@ -0,0 +1,342 @@
+package createsetters
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/kio/kioutil"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+var _ kio.Filter = &CreateSetters{}
+
+// CreateSetters creates a comment for the resource fields which
+// contain the same value as setter value
+type CreateSetters struct {
+	// Setters holds the user provided values for simple map setters
+	ScalarSetters []ScalarSetter
+
+	// ArraySetters holds the user provided values for array setters
+	ArraySetters []ArraySetter
+
+	// Results are the results of adding setter comments
+	Results []*Result
+
+	// filePath file path of resource
+	filePath string
+}
+
+// ScalarSetter stores name and value of the map setter
+type ScalarSetter struct {
+	// Name is the name of the setter
+	Name string
+
+	// Value is the value of the field to which setter comment is added.
+	Value string
+}
+
+// ArraySetter stores name and values of the array setter
+type ArraySetter struct {
+	// Name is the name of the setter
+	Name string
+
+	// Values are the values of the field to which setter comment is added.
+	Values []string
+}
+
+// Result holds result of create-setters operation
+type Result struct {
+	// FilePath is the file path of the matching value
+	FilePath string
+
+	// FieldPath is field path of the matching value
+	FieldPath string
+
+	// Value is the value of the field to which setter comment is added.
+	Value string
+
+	// LineComment of the matching value
+	Comment string
+}
+
+// CompareSetters is to sort the setter values
+type CompareSetters []ScalarSetter
+
+func (a CompareSetters) Len() int {
+	return len(a)
+}
+
+func (a CompareSetters) Less(i, j int) bool {
+	return !strings.Contains(a[i].Value, a[j].Name)
```

Comment:
Explain why you are sorting and what you are achieving with example in comments.

Reply:
Sure